### PR TITLE
Gc cpp

### PR DESCRIPTION
**Pull request by lausdahl**
added cpp code gen and changed data writer to use vargs

### CODE REVIEW

**Comment by lausdahl** [comment]
@CThuleHansen any ideas to why I got the diff error?

File: codegen/mabl2cpp/src/main/resources/org/intocps/maestro/codegen/mabl2cpp/libs/DataWriter.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+#include <iostream>
+#include <iostream>
+#include <fstream>
+
+#define DataWriterConfig void *
```
Comment (CThuleHansen) [inline codegen/mabl2cpp/src/main/resources/org/intocps/maestro/codegen/mabl2cpp/libs/DataWriter.h:6]: These looks weird. Why not typedef?

File: codegen/mabl2cpp/src/main/resources/org/intocps/maestro/codegen/mabl2cpp/libs/DataWriter.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    DataWriterConfig writeHeader( int count,std::string[]);
+
+    void close();
+
+    private:
+    std::ofstream myfile;
+};
+
+#define DataWriter DataWriterImpl*
```
Comment (CThuleHansen) [inline codegen/mabl2cpp/src/main/resources/org/intocps/maestro/codegen/mabl2cpp/libs/DataWriter.h:20]: Typedef?

**Review by CThuleHansen — COMMENTED** [review]
Is there a reason for all the commented code? Or because we keep this work as prototypeish for now?

**Pull request merged**